README.md

```diff
@@ -140,6 +140,35 @@ float
 
 </table>
 
+## Coding rules
+
+We define a list of coding rules for the OpenCL-code so that facilitate using it from multiple frameworks.
+
+### Source and destination images/buffers.
+
+There are many image processing filters which just take one source image and produce one destination image. Those should be simply called `src` and `dst`. In case multiple source and/or destination images appear in the kernel, those should be called `src0`, `src1`, ...
+
+### nD-kernels (n<=3)
+
+We can write image processing kernels that can work with 1, 2 and 3-dimensional images. Therefore, it is necessary to always write code like this:
+```
+const int x = get_global_id(0);
+const int y = get_global_id(1);
+const int z = get_global_id(2);
+
+POS_src_TYPE pos = POS_src_INSTANCE(x,y,z,0);
+
+flooat value = READ_src_IMAGE(src, sampler, pos).x;
```

Comment:
small typo

Reply:
Where? Use the suggestion button to fix it ;-)

![image](https://user-images.githubusercontent.com/12660498/144031798-512b88c9-4a2f-4dbf-9802-ecba6323f57a.png)

Reply:
```suggestion
float value = READ_src_IMAGE(src, sampler, pos).x;
```
still discovering these functionality !

Reply:
cooool, I didn't see this one :-) Thanks!